### ISSUE TITLE
[Bug] Fused silu_and_mul post-quant launchers build invalid CUDA launch configs (block > 1024 threads; zero-block grid on empty batch)

### ISSUE THREAD

**Issue by lcz1998** [sgl-project/sglang#32065]
# [Bug] Fused silu_and_mul post-quant launchers build invalid CUDA launch configs (block > 1024 threads; zero-block grid on empty batch)

## Checklist

- [x] I searched for related issues and did not find a duplicate.
- [x] The bug is reproducible on the latest `main`.

## Describe the bug

The launchers of the fused SiLU-and-mul + FP8 quantization kernels in
`python/sglang/jit_kernel/csrc/deepseek_v4/silu_and_mul_masked_post_quant.cuh`
(`SiluAndMulMaskedPostQuantKernel` and `SiluAndMulContigPostQuantKernel`)
have two launch-configuration problems:

**1. Block size is unchecked against the 1024-threads-per-block limit.**

```cpp
    const auto num_threads = hidden_dim / 8;
    RuntimeCheck(num_threads % device::kWarpThreads == 0);
    ...
    LaunchKernel(num_tokens * topk, num_threads, device.unwrap())
```

`num_threads` is derived directly from the (intermediate) hidden dim. Any
MoE with `moe_intermediate_size > 8192` yields more than 1024 threads per
block and the launch fails with `cudaErrorInvalidConfiguration` at runtime,
with no hint that the kernel simply does not support that shape. The
neighboring `SiluAndMulClampKernel` in the same file already carries this
check (`RuntimeCheck(num_threads <= 1024, "out_dim too large ...")`); the
two quant launchers do not.

**2. Empty batches launch a zero-block grid.**

`num_tokens == 0` (masked variant: also `topk == 0`) still calls
`LaunchKernel(0, num_threads, ...)`. A grid dimension of 0 is an invalid
CUDA launch configuration and fails with `cudaErrorInvalidConfiguration`.

Empty batches are not hypothetical: with `--enable-dp-attention`, a DP rank
that receives no requests in a scheduling step still executes the MoE path
with zero tokens, and crashes here when the fused activation-quant path is
enabled (e.g. via `SGLANG_OPT_USE_JIT_EP_ACTIVATION` /
`SGLANG_OPT_FIX_MEGA_MOE_MEMORY` on the DeepGEMM runner).

## Reproduction

Zero-token repro (direct kernel call):

```python
import torch
from sglang.jit_kernel.deepseek_v4 import silu_and_mul_masked_post_quant  # jit entry
# any tensors with num_tokens == 0 -> cudaErrorInvalidConfiguration
```

End-to-end: run a DP-attention deployment where one rank goes idle for a
step with the JIT EP activation path enabled; the idle rank crashes in the
fused quant launcher.

## Expected behavior

- Unsupported hidden dims should fail with a clear `RuntimeCheck` message
  instead of a raw CUDA launch error.
- Empty batches are a valid input (nothing to do) and should be a no-op,
  consistent with how other SGLang kernels handle `num_tokens == 0`.

## Proposed fix

```cpp
    const auto num_threads = hidden_dim / 8;
    RuntimeCheck(num_threads > 0 && num_threads <= 1024, "hidden_dim/8 exceeds CUDA block limit");
    RuntimeCheck(num_threads % device::kWarpThreads == 0);
    ...
    if (num_tokens == 0 || topk == 0) {
      return;
    }
```

(and the `num_tokens == 0` early-return for the contiguous variant).
The params struct is still validated before the early return, so shape
errors are reported even for empty batches. PR: (linked below)

## Environment

- SGLang: main (verified present on 977ea33, 2026-07-22; also on 2dd449c/c9303)
- GPU: 8x H800, CUDA 12.x
- Trigger: DP attention with an idle rank + JIT EP activation fused quant path
